exercise_Melissa.Rmd

```diff
@@ -0,0 +1,106 @@
+
+    Follow [these guidelines for reviewing a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request), give some meaningful feedback, and when you're done 'Approve' the pull request (do NOT close or merge).
+
+## The coding exercise:
+
+Work through the steps below to complete this exercise (to the best of your ability) and then save, add, commit and push this file with your coding edits to your forked repo and follow the instructions above for completing the assignment via creating and reviewing a pull request.
+
+Load in the tidyverse (only package(s) needed for this exercise:
+
+```{r}
+library(tidyverse)
+library(dplyr)
+```
+
+Read in the 'kenya_county_climates.csv' in the 'data/' folder and assign it to an environmental variable (Hint: use 'read_csv()').
+
+```{r}
+kenya_county_climates <- read.csv("/Users/melissamorales/Desktop/Week-1-Intro/data/kenya_county_climates.csv")
+```
+
+Write a chain of operations that does the following:
+
+1.  `select()` just the columns that have "Precipitation" in the name (so we are only interested in the precipitation data), AND the 'County' column.
+
+2.  `filter()` all the rows where 'Precipitation_585_90' is **greater than** the mean of all county precipitation values.
+
+3.  `pivot_longer()` all the precipitation columns (so everything *except* 'County'), so we have one column for all precip values and one column with the value's associated variables.
+
+```{r}
+# select for County and Precipitation
+
+select(kenya_county_climates, contains("Precipitation"), "County") %>%
+filter(Precipitation_585_90> mean(Precipitation_hist)) %>%
+pivot_longer(cols = (-County))
+
+df$new_col
+new_col %>%
+  n_distinct("County")
+```
+
+Question: How many counties have predicted precip values greater than the total average for all counties?
+28
+
+Using the new dataset created above, create a bar plot of precipitation for each time frame for each county using `facet_wrap()`
+
+```{r}
+df %>%
+ggplot(mapping = aes(x=name, y=value)) + 
+  geom_col() + 
+  facet_wrap(~County)
+```
+
+CHALLENGE:
+
+This is messy though since we have so many counties. Instead, use `map()` to create an individual bar plot for each County (hint: create a vector of county names from the data, and building off the ggplot code you made above, add in a line to `filter()` out each county (i.e., .x) before creating the plot. AND, no more need to `facet_wrap()` here).
+
+There are many ways to do this!
+
+```{r}
+library(purrr)
+```
+
+```{r}
+# Mean precip
+county_means <- kenya_county_climates %>%
+  group_by(County) %>%
+  summarize(mean_precipitation = mean(Precipitation_hist))
+
+# Bar plot
+ggplot(county_means, aes(x = County, y = mean_precipitation)) +
+  geom_bar(stat = "identity", fill = "steelblue") +
+  labs(title = "Mean Precipitation by County",
+       x = "County",
+       y = "Mean Precipitation") +
+  theme_minimal()
+
```

Comment:
I think this should all work, and I did not end up finishing this problem, but I think the exercise wanted us to try to us map(). This is the code from the answer key 

counties <- unique(data_filtered$County)

plots <-
  map(counties,
      ~ data_filtered %>% filter(County == .x) %>% ggplot(aes(x = name, y = value)) + geom_col())